lib/jxl/encode.cc

```diff
@@ -433,10 +444,18 @@ JxlEncoderStatus JxlEncoderOptionsSetInteger(JxlEncoderOptions* options,
       // alternatively, in the cjxl binary like now)
       options->values.cparams.lossy_palette = (value == 1);
       return JXL_ENC_SUCCESS;
+    case JXL_ENC_OPTION_COLOR_TRANSFORM:
+      if (value < -1 || value > 2) return JXL_ENC_ERROR;
+      if (value == -1) {
+        options->values.cparams.color_transform = jxl::ColorTransform::kXYB;
```

Comment:
As discussed elsewhere, default setting probably shouldn't happen here in the first place (but only when actual encoding starts). In this case, for example, kXYB is a good default for lossy while for lossless kNone has to be used and it would not make sense to use kXYB by default.

Reply:
In CompressParams, color_transform is initialized as `ColorTransform color_transform = ColorTransform::kXYB`, so if the implementation here sets it to kXYB, it does exactly the same as before. It's not possible that this MR can introduce the problem of wrong default color transform value.

So this MR does not change any behavior. If we need a refactoring for default values, let's do it in a different MR. What do you think?

Reply:
Yes, let's do it in a follow-up MR. I think it's best to set things to -1 by default in the api options, and to initialize cparams just before actual compression starts, so defaults can be made sensible (and we can e.g. avoid the problem that applications think they're doing lossless but it's actually XYB).